tasks/managed/filter-already-released-images/filter-already-released-images.yaml

```diff
@@ -0,0 +1,341 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: filter-already-released-images
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: |- 
+    Tekton task to filter out images from a snapshot that have already been released.
+    This task checks target registries to determine if push-snapshot has completed successfully
+    for each component by validating that ALL required tags exist with the correct digest.
+    Components that are fully released (all tags present) are filtered out before EC validation.
+    
+    Tag-level validation ensures complete releases and prevents filtering components with 
+    partial tag pushes. A component is only filtered if at least one repository has ALL
+    required tags pointing to the correct digest.
+    
+    The task overwrites the original snapshot file in place with a filtered version 
+    containing only unpublished or partially published images.
+    
+    This task must run AFTER apply-mapping since it needs the mapped target repositories
+    and their required tags from the enriched snapshot stored in trusted artifacts
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the Snapshot spec in the data workspace
+      type: string
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+    - name: ociStorage
+      description: The OCI repository where the Trusted Artifacts are stored
+      type: string
+      default: "empty"
+    - name: ociArtifactExpiresAfter
+      description: |-
+        Expiration date for the trusted artifacts created in the OCI repository. 
+        An empty string means the artifacts do not expire
+      type: string
+      default: "1d"
+    - name: trustedArtifactsDebug
+      description: Flag to enable debug logging in trusted artifacts. Set to a non-empty string to enable
+      type: string
+      default: ""
+    - name: orasOptions
+      description: oras options to pass to Trusted Artifacts calls
+      type: string
+      default: ""
+    - name: sourceDataArtifact
+      type: string
+      description: Location of trusted artifacts to be used to populate data directory
+      default: ""
+    - name: dataDir
+      description: The location where data will be stored
+      type: string
+      default: /var/workdir/release
+    - name: taskGitUrl
+      type: string
+      description: The url to the git repo where the release-service-catalog tasks to be used are stored
+      default: https://github.com/konflux-ci/release-service-catalog.git
+    - name: taskGitRevision
+      type: string
+      description: The revision in the taskGitUrl repo to be used
+  results:
+    - name: filteredCount
+      description: Number of components filtered out as already released
```

Comment:
Why do we want this information in other tasks?

Reply:
I don't think it breaks anything, I just don't really see the value - it is a task result, not a pipeline result, so it isn't ever exposed to users. It is just in the task log, which would be true whether or not it is a task result (you have to go look for the echo in the log regardless)

Reply:
np, as I mentioned, if you think it would be better to remove it, there is no problem. It was just easier to read the results for the e2e test (then parse logs), and it doesn't hurt anything (minimal overhead)
removed